docker/configure_workers_and_start.py

```diff
@@ -476,6 +478,39 @@ def generate_worker_files(
     # which exists even if no workers do.
     healthcheck_urls = ["http://localhost:8080/health"]
 
+    # Expand worker_type multiples if requested in shorthand(e.g. worker:2). Checking
+    # for not an actual defined type of worker is done later.
+    # Checking performed:
+    # 1. if worker:2 or more is declared, it will create additional workers up to number
+    # 2. if worker:1, it will create a single copy of this worker as if no number was
+    #   given
+    # 3. if worker:0 is declared, this worker will be ignored. This is to allow for
+    #   scripting and automated expansion and is intended behaviour.
+    # 4. if worker:NaN or is a negative number, it will error and log it.
+    new_worker_types = []
+    for worker_type in worker_types:
+        if ":" in worker_type:
```

Comment:
I'd probably do something like

```python
pieces = worker_type.split(":", maxsplit=1)
worker_name = pieces[0]
try:
    worker_count = int(pieces[1] if len(pieces) >= 2 else 1)
except ValueError:
    error(f"Bad number in worker count for '{worker_type}': '{pieces[1]}'")
```
 
It's not exactly difficult to read but `x`, `y` don't really help.

Reply:
I was trying to stay away from `try: except` usage and just make use of the `error()` helper. Also, this would default it to `1` if it borks? Wouldn't that bypass the point of stopping everything so it could be fixed?

Reply:
I played around with this a bit, as I'm not used to using `try: except` stuff and I think I like it. However, I will have to change the `split_string_and_split()` to accept an optional `maxsplit` argument/parameter as I want to keep the stripping whitespace functionality. I did also remove the conditional statement in the `int()` casting, as it's splitting the string exactly one time so will never have more than 2 and even if it ends up as only one value, it will still fail at the `try:`. Does that make sense?

(I'll change the other places similar processing is done too, like for the name. I may need help with that one)

Reply:
Notable but not obvious from the extract: `error` quits the script with an error code, so it won't escape past the ValueError.

`try...except` for handling this type of error is basically the norm / idiom in Python, moreso than in other languages. As they say: when in Rome, do as the Romans do. :P
(Well, to be fair, this would not be unfair in most languages since it really is an error condition, but in Python it's considered fair to use exceptions for flow control in non-exceptional cases, which is frowned upon in other languages!)

Reply:
A lot of the problems I check for won't be problems until after this script has finished and Synapse is in the process of starting up. Like a 'name' containing a space would cause a `FileNotFound` error. Perhaps I could change the `error()` to be `raise ValueError()` or `raise ConfigError()`?

Reply:
look at the code for `error`: it quits immediately :)

Reply:
Sorry, I meant for the other places. This one I'm good with, but the rest would(I'm pretty sure) pass a `try`.